src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
+                return throwableClass.cast(t);
```

Comment:
@marcphilipp Sure, I'll add them to this PR.

@kcooney I'm pretty sure it's an IntelliJ inspection. We could also just suppress the warning on `expectThrows`, but I really like having `assertThrows` for discoverability and `expectThrows` to avoid muddying the waters of the current convention (currently all `assert` methods are void, right?). I'm not sure what the googletest convention has to do with anything. I can just as easily point at the [expect method](https://doc.rust-lang.org/std/option/enum.Option.html#method.expect) in Rust's Option type, which makes an assertion en passant and then returns a value.